src/utilities/time.ts

```diff
@@ -91,28 +92,31 @@ export function formatDate(date: Date): string {
 }
 
 export function getDateBounds(
-  data: GraphTimelineNode[],
+  data: TimelineData,
   minimumTimeSpan: number = 0,
   isRunning: boolean = false,
 ): { min: Date, max: Date, span: number } {
+  let minStartTime: number | null = null
+  let maxEndTime: number | null = isRunning ? new Date().getTime() : null
 
-  const defaultMaxEndTime = isRunning ? new Date().getTime() : null
-
-  const [minStartTime, maxEndTime] = data.reduce<[number | null, number | null]>(([min, max], { start, end }) => {
+  data.forEach(({ start, end }) => {
     const startTime = start?.getTime() ?? null
     const endTime = end?.getTime() ?? null
 
-    if (startTime && min) {
-      min = Math.min(min, startTime)
+    if (startTime && minStartTime) {
+      minStartTime = Math.min(minStartTime, startTime)
+    } else {
+      minStartTime = startTime
     }
 
-    if (endTime && max) {
-      max = Math.max(max, endTime)
+    if (endTime && maxEndTime) {
+      maxEndTime = Math.max(maxEndTime, endTime)
+    } else {
+      maxEndTime = endTime
     }
+  })
 
-    return [min ?? startTime, max ?? endTime]
-  }, [null, defaultMaxEndTime])
-
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
Sorry should have added an explanation to the comment. Typescript doesn't follow `let` very well sometimes. Here I couldn't get it to understand that it could in fact be null